pkg/controller/user/create.go

```diff
@@ -41,87 +35,113 @@ func (c *Controller) HandleCreate() http.Handler {
 		}
 		flash := controller.Flash(session)
 
-		membership := controller.MembershipFromContext(ctx)
-		if membership == nil {
+		currentMembership := controller.MembershipFromContext(ctx)
+		if currentMembership == nil {
 			controller.MissingMembership(w, r, c.h)
 			return
 		}
-		if !membership.Can(rbac.UserWrite) {
+		if !currentMembership.Can(rbac.UserWrite) {
 			controller.Unauthorized(w, r, c.h)
 			return
 		}
+		currentRealm := currentMembership.Realm
+		currentUser := currentMembership.User
 
-		currentRealm := membership.Realm
-		currentUser := membership.User
+		user := &database.User{}
+		userMembership := &database.Membership{}
 
-		// Requested form, stop processing.
 		if r.Method == http.MethodGet {
-			c.renderNew(ctx, w)
+			c.renderNew(ctx, w, user, userMembership)
 			return
 		}
 
-		var form FormData
-		if err := controller.BindForm(w, r, &form); err != nil {
-			flash.Error("Failed to process form: %v", err)
-			c.renderNew(ctx, w)
+		if err := bindCreateForm(r, currentMembership, user, userMembership); err != nil {
+			user.AddError("", err.Error())
+			w.WriteHeader(http.StatusUnprocessableEntity)
+			c.renderNew(ctx, w, user, userMembership)
 			return
 		}
 
 		// See if the user already exists by email - they may be a member of another
 		// realm.
-		user, err := c.db.FindUserByEmail(form.Email)
-		if err != nil {
-			if !database.IsNotFound(err) {
-				controller.InternalError(w, r, c.h, err)
+		existing, err := c.db.FindUserByEmail(user.Email)
+		if err != nil && !database.IsNotFound(err) {
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+		if existing != nil && existing.ID != 0 {
+			user.ID = existing.ID
+		}
+
+		// Create or update user.
+		if err := c.db.SaveUser(user, currentUser); err != nil {
+			if database.IsValidationError(err) {
+				w.WriteHeader(http.StatusUnprocessableEntity)
+				c.renderNew(ctx, w, user, userMembership)
```

Comment:
Can we add `show User validation errors` to the description too

Reply:
Done